python/GafferSceneUI/TransformToolUI.py

```diff
@@ -161,29 +161,46 @@ def __update( self, *unused ) :
 			return
 
 		toolSelection = self.__tool.selection()
-		selectedPaths = GafferSceneUI.ContextAlgo.getSelectedPaths( self.context() )
 
 		if len( toolSelection ) :
 
-			self.__infoRow.setVisible( True )
-			if len( toolSelection ) == 1 :
+			# Get unique edit targets and warnings
+
+			editTargets = set()
+			warnings = set()
+			allEditable = True
```

Comment:
As we have `selectionEditable` on the tool, would it make sense to use that instead? I _think_ you could then make `warnings` and `editTargets` be build from list comprehensions and slightly simplify visibility management of `__infoRow` below.

Reply:
Gave this a bash in 9e8b87e. I had actually started off with set comprehensions and then rewrote it the other way when I realised I wanted to compute `allEditable` too - thanks for pointing out that it was available all along!

Reply:
I've deliberately left the visibility management of `__infoRow` explicit and slightly repetitive so that each conditional block is the master of its own destiny.